### DISCUSSION TITLE
Multiline class integration

### DISCUSSION THREAD

**Discussion by yanntouil** [moalamri/vscode-inline-fold#38] [Ideas]
Hello,
I use your extension with Tailwind, to make my code more readable and I'm a big fan.
A problem occurs when using classNames() the fold multiline doesn't work properly :
That
`
   <a className={vClass(
      'flex items-center p-4 gap-4 text-stone-700 border-b border-stone-100'
   )}>Lien 2</a>
`
Give me
`
   <a className={•••
      'flex items-center p-4 gap-4 text-stone-700 border-b border-stone-100'
   )}>Lien 2</a>
`
I tested with another regex that works well but the multiline remains
`
(class|className)=[`'{"]([^<]+)[`'"}]
`
This feature would make your extension just perfect.

**Comment by moalamri** [comment]
> 

UPDATE: [Check my other comment](https://github.com/moalamri/vscode-inline-fold/discussions/38#discussioncomment-3329075)

Hi.
I think the name gives a good hint 😄 .
The extension can't fold multiline because it's basically a decorator not a real API (VS Code doesn't support inline-folding and it's been in the backlog for 4 years).
If you were to fold multiline with this extension then you will have empty lines due to being decorated to be hidden and not actually folded 😞.
But the good news is there's an extension that could do what you are aiming for, but I don't really know how it is going to blend with my extension. You can get it from here [Explicit Folding](https://marketplace.visualstudio.com/items?itemName=zokugun.explicit-folding)

**Comment by moalamri** [comment]
@yanntouil 
Great news!!

Update VS Code to the latest version 1.70.0
They added a new feature where you can create custom folding for multiline.
You just have to select the region and run the command.

![image](https://user-images.githubusercontent.com/964077/183016270-b5f0263d-9706-4f3a-803f-b21b74c4eabe.gif)

**Reply by yanntouil** [reply]
After some testing, no feature makes life easier than your extension. I'll just exclude ClassNames from the regex to use it only on strings.
Anyway, a big thank you for your excellent work on this extension and for the time you gave me.
